virttest/virt_vm.py

```diff
@@ -1002,7 +1012,10 @@ def verify_kernel_crash(self):
                 return
             match = re.search(panic_re, data, re.DOTALL | re.MULTILINE | re.I)
             if match:
-                raise VMDeadKernelCrashError(match.group(0))
+                if self.is_dead():
```

Comment:
`is_dead` can only check the qemu process, not the guest status. when the guest crashes, if qemu can report `guest-panicked` status, then we can use it, but if not, there is not a good idea the check the guest status.

Reply:
> `is_dead` can only check the qemu process, not the guest status. when the guest crashes, if qemu can report `guest-panicked` status, then we can use it, but if not, there is not a good idea the check the guest status.

Hi @PaulYuuu 
You's concern makes sense. 
I think we should give the definition of the `VMDead` of `VMDeadKernelCrashError`.
What's the meaning of `VMDead`? Does it relate to the VM class?
In another way, how do we define the VM's dead status here?
CC @zhenyzha

Reply:
Hello guys,

For me, it's relatively accurate to use it for checking in `virt_vm`.
We can refer to `verify_alive(self)`, its judgment is also through `self.is_dead()`
https://github.com/avocado-framework/avocado-vt/blob/master/virttest/virt_vm.py#L699

If we really want to explicitly define `VMDead` from the abstraction layer, 
I think overall refactoring may be needed here.
Of course, if there is a better way of judging, I am also happy to correct it.

Reply:
I think you should check `verify_alive` in qemu_vm.py and libvirt_vm.py, this method in virt_vm.py never be used for tp-qemu and tp-libvirt.